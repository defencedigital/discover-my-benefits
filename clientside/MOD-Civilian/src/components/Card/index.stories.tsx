import React from 'react';
import { storiesOf } from '@storybook/react';
import { Card } from '@components';

storiesOf('Card', module).add('Card', () => {
  const data = {
    name: 'Category Name',
    slug: '/lorem-ipsum',
    headerImage: [
      {
        url: 'https://mod-squidex-dev.cloudapps.digital/api/assets/f4936a47-f9c8-47bf-93e7-0c665daba0b6?version=0',
      },
    ],
    popularBenefits: [
      {
        flatData: {
          name: 'Flexible working',
          slug: '/flexible-working',
        },
        id: 'asdf',
      },
    ],
  };
  return (
    <React.Fragment>
      <Card data={data} />
    </React.Fragment>
  );
});
