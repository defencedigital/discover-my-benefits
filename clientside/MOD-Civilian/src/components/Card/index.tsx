import * as React from 'react';
import { Link, withPrefix } from 'gatsby';

export interface ICardComponentProps {
  data: {
    name: string;
    slug: string;
    headerImage: [
      {
        url: string;
      },
    ];
    popularBenefits: [
      {
        flatData: {
          name: string;
          slug: string;
        };
        id: string;
      },
    ];
  };
}

const PopularBenefitList = ({ url, benefits }) => {
  return (
    <div className="flex flex-col justify-start flex-grow">
      <h4 className="font-bold pb-1">Popular Benefits:</h4>
      {benefits.map(item => {
        return (
          <Link
            data-ga-category="category-card-popular-benefit-link"
            data-ga-action="click"
            data-ga-label={item.flatData.name}
            className="underline text-sm pb-1 hover:text-primary"
            key={item.id}
            to={`/${url}/${item.flatData.slug}`}
          >
            {item.flatData.name}
          </Link>
        );
      })}
    </div>
  );
};

function Card({ data }: ICardComponentProps) {
  const { slug, name, headerImage, popularBenefits } = data;
  const { url } = headerImage[0];

  return (
    <div className="w-full sm:w-1/2 lg:w-1/3 px-4 mb-4">
      <div className="rounded border-solid border border-gray-300 overflow-hidden flex-1 flex flex-col h-full justify-start">
        <Link to={`/${slug}`} data-ga-category="category-card" data-ga-action="click" data-ga-label={name} className="w-full">
          <picture className="w-full object-cover print:hidden">
            <source srcSet={`${url}&width=300`} type="" />
            <img className="w-full object-cover h-full" src={`${url}&width=300`} alt={name} />
          </picture>
        </Link>
        <div className="p-2 h-full flex flex-col justify-start">
          <Link
            data-ga-category="category-card-title-link"
            data-ga-action="click"
            data-ga-label={name}
            className="hover:underline hover:text-primary"
            to={`/${slug}`}
          >
            <h3 className="font-bold text-xl pb-2 underline">{name}</h3>
          </Link>
          {popularBenefits && <PopularBenefitList url={slug} benefits={popularBenefits} />}
          <Link
            data-ga-category="category-card-more-link"
            data-ga-action="click"
            data-ga-label={name}
            className="underline text-primary hover:text-secondary flex pt-3 text-sm "
            to={`/${slug}`}
            aria-label={`See more links under ${name}`}
          >
            more...
          </Link>
        </div>
      </div>
    </div>
  );
}

export default Card;
