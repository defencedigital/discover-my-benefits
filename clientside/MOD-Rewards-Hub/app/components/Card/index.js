import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardHeader, CardImg, CardSubtitle, CardText, CardTitle, Row, Col } from 'reactstrap';
import { statusClasses, statuses } from '../../containers/Benefits/helpers';
import MissingIcon from '../../images/svg/question.svg';
import EligibleIcon from '../../images/svg/tick.svg';
import SheildIcon from '../../images/svg/Shield_Icon.svg';
import GatewayIcon from '../../images/svg/Gateway_Icon.svg';

class C extends React.PureComponent {
  handleEligibilityIconsRender(eligibility) {
    if (eligibility.Eltotal > 0 && eligibility.Mtotal > 0 && eligibility.Eltotal !== eligibility.Mtotal) {
      return (
        <div className="card-eligibility-indicators">
          <span>
            <EligibleIcon /> Eligible for {eligibility.Eltotal} benefit(s){' '}
          </span>
          <span>
            <MissingIcon /> {eligibility.Mtotal}
          </span>
        </div>
      );
    }
    if (eligibility.Eltotal > 0 && eligibility.Mtotal > 0 && eligibility.Eltotal === eligibility.Mtotal) {
      return (
        <div className="card-eligibility-indicators">
          <span>
            <EligibleIcon /> Eligible for {eligibility.Eltotal} benefit(s){' '}
          </span>
          <span>
            <MissingIcon /> {eligibility.Mtotal}
          </span>
        </div>
      );
    }
    if (eligibility.Eltotal <= 0 && eligibility.Mtotal > 0) {
      return (
        <div className="card-eligibility-indicators">
          <span>
            <MissingIcon /> Missing info for {eligibility.Mtotal} benefit(s)
          </span>
        </div>
      );
    }
    if (eligibility.Eltotal > 0 && eligibility.Mtotal <= 0) {
      return (
        <div className="card-eligibility-indicators">
          <span>
            <EligibleIcon /> Eligible for {eligibility.Eltotal} benefit(s){' '}
          </span>
        </div>
      );
    }
    return (
      <div className="card-eligibility-indicators">
        <span className="card-eligibility-text">Read to find out if you are eligible</span>
      </div>
    );
  }

  render() {
    const {
      className,
      onClick,
      status,
      link,
      openLinkInNewWindow,
      title,
      image,
      text,
      eligibility,
      subtitle,
      hideTextOnXs,
      tags,
      twoCol,
      modnetLink,
      gatewayLink,
    } = this.props;

    const externalLink = link.indexOf('http') > -1;
    return (
      <Card
        tag="article"
        id={`card-${title.toLowerCase().replace(/ /g, '-')}`}
        className={className}
        style={{ height: '100%' }}
        onClick={onClick}
        data-component="card"
      >
        {status !== null && (
          <CardHeader className={`card-header-${statusClasses[status]}`}>
            <span>{statuses[status]}</span>
          </CardHeader>
        )}
        {gatewayLink === true && (
          <CardHeader className="card-header-link">
            <span>
              <GatewayIcon /> Defence Gateway
            </span>
          </CardHeader>
        )}
        {modnetLink === true && (
          <CardHeader className="card-header-link">
            <span>
              <SheildIcon /> DEFnet access only
            </span>
          </CardHeader>
        )}

        <div className="card-inr">
          {twoCol && image && (
            <Row>
              <Col xs="12" md="6" lg="6">
                <CardImg top width="100%" loading="lazy" src={image} alt={title} />
              </Col>
              <Col xs="12" md="6" lg="6">
                <div
                  className={`card-body card-body-has-sibling ${hideTextOnXs ? 'adjust-padding-mobile' : ''}`}
                >
                  {eligibility && this.handleEligibilityIconsRender(eligibility)}
                  {title && !externalLink && (
                    <Link
                      className="card-link"
                      target={openLinkInNewWindow ? '_blank' : null}
                      to={link}
                      onClick={() => {}}
                      data-ga-category="card"
                      data-ga-action="click"
                      data-ga-label={title}
                    >
                      <CardTitle
                        data-ga-category="card"
                        data-ga-action="click"
                        data-ga-label={title}
                        tag="h3"
                        className="h4"
                      >
                        {title}
                      </CardTitle>
                    </Link>
                  )}
                  {subtitle && <CardSubtitle>{subtitle}</CardSubtitle>}
                  {text && <CardText className={hideTextOnXs ? 'hideTextOnXs' : ''}>{text}</CardText>}
                  <div className="card-tags">
                    {tags &&
                      this.props.tags.map(tag => (
                        <span
                          key={tag.name}
                          className={`card-tag ${tag.globalFamilyTag ? 'card-tag--fam' : ''}`}
                        >
                          {tag.name}
                        </span>
                      ))}
                  </div>
                </div>
              </Col>
            </Row>
          )}

          {!twoCol && image && <CardImg top width="100%" loading="lazy" src={image} alt={title} />}
          {!twoCol && !externalLink && (
            <div className={`card-body card-body-has-sibling ${hideTextOnXs ? 'adjust-padding-mobile' : ''}`}>
              {eligibility && this.handleEligibilityIconsRender(eligibility)}
              {title && (
                <Link
                  className="card-link"
                  target={openLinkInNewWindow ? '_blank' : null}
                  to={link}
                  onClick={() => {}}
                  data-ga-category="card"
                  data-ga-action="click"
                  data-ga-label={title}
                >
                  <CardTitle
                    data-ga-category="card"
                    data-ga-action="click"
                    data-ga-label={title}
                    tag="h3"
                    className="h4"
                  >
                    {title}
                  </CardTitle>
                </Link>
              )}
              {subtitle && <CardSubtitle>{subtitle}</CardSubtitle>}
            </div>
          )}

          {externalLink && (
            <a
              className="card-link"
              target={openLinkInNewWindow ? '_blank' : null}
              href={link}
              onClick={() => {}}
              data-ga-category="card"
              data-ga-action="click"
              data-ga-label={title}
            >
              {image && <CardImg top width="100%" loading="lazy" src={image} alt={title} />}

              <div
                className={`card-body card-body-has-sibling ${hideTextOnXs ? 'adjust-padding-mobile' : ''}`}
              >
                {eligibility && this.handleEligibilityIconsRender(eligibility)}
                {title && (
                  <CardTitle
                    data-ga-category="card"
                    data-ga-action="click"
                    data-ga-label={title}
                    tag="h3"
                    className="h4"
                  >
                    {title}
                  </CardTitle>
                )}
                {subtitle && <CardSubtitle>{subtitle}</CardSubtitle>}
              </div>
            </a>
          )}

          {!link && onClick && (
            <>
              {image && <CardImg top width="100%" loading="lazy" src={image} alt={title} />}
              <div className="card-body card-body-has-sibling">
                {title && (
                  <CardTitle
                    data-ga-category="card"
                    data-ga-action="click"
                    data-ga-label={title}
                    tag="h3"
                    className="h4"
                  >
                    {title}
                  </CardTitle>
                )}
                {subtitle && <CardSubtitle>{subtitle}</CardSubtitle>}
              </div>
            </>
          )}
          {!twoCol && (
            <div className="card-body-sibling">
              {text && <CardText className={hideTextOnXs ? 'hideTextOnXs' : ''}>{text}</CardText>}
              <div className="card-tags">
                {tags &&
                  this.props.tags.map(tag => (
                    <span key={tag.id} className={`card-tag ${tag.globalFamilyTag ? 'card-tag--fam' : ''}`}>
                      {tag.name}
                    </span>
                  ))}
              </div>
            </div>
          )}
        </div>
      </Card>
    );
  }
}

C.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  text: PropTypes.string,
  hideTextOnXs: PropTypes.bool,
  image: PropTypes.string,
  link: PropTypes.string,
  openLinkInNewWindow: PropTypes.bool,
  twoCol: PropTypes.bool,
  className: PropTypes.string,
  status: PropTypes.number,
  onClick: PropTypes.func,
  tags: PropTypes.array,
  eligibility: PropTypes.object,
  modnetLink: PropTypes.bool,
  gatewayLink: PropTypes.bool,
};

C.defaultProps = {
  status: null,
};

C.displayName = 'Card';

export default C;
