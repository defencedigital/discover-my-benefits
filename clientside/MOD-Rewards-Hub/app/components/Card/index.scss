.card {
  display: block !important;
  text-align: left;
}

.active.card {
  border-color: $brand-primary;

  .card-header {
    border-color: $brand-primary;
  }
}

.not-eligible.card {
  // background-color: $gray-lighter;
  color: $brand-primary;

  .card-link {
    color: $brand-primary;

    &:hover {
      color: $brand-secondary-alpha;
    }
  }

  .card-header {
    color: $brand-primary;
  }
}

.not-eligible.active.card {
  // background-color: $gray-lighter;
  color: $brand-primary;
  border-color: $brand-primary;

  .card-header {
    border-color: $brand-primary;
    color: $brand-primary;
  }
}

.disabled.active.card {
  border-color: $brand-primary;

  .card-header {
    border-color: $brand-primary;
  }

}

.card-header-link svg{
  width: 20px;
  position: absolute;
  left: 9px;
  top: 8px;
}

.card-link {
  &:hover {
    cursor: pointer;

    .card-title {
      text-decoration: underline;
    }
  }

  &.disabled {
    text-decoration: none;
    //color: $black;
    cursor: default;

    &:hover {
      text-decoration: none;
      //color: $black;
      cursor: default;

      .card-title {
        text-decoration: none;
        //color: $black;
        cursor: default;
      }
    }
  }
}

.card-text {
  font-size: 1.4rem;
  line-height: 2.0rem;
}

.card-body {
  display: block;
  padding: 16px 13px;
  position: relative;

  &.adjust-padding-mobile {
    @include media-breakpoint-down(xs) {
      padding: 8px;
    }
  }
}

.card-body-has-sibling {
  padding-bottom: 0;
}

.card-body-sibling {
  padding: 0 13px 13px 13px;

  .hideTextOnXs {
    @include media-breakpoint-down(xs) {
      display: none;
    }
  }
}

.card-header {
  position: relative;
  background-color: $gray-light;
  padding: 13px;
  min-height: 40px;
}

.card-header-eligible,
.card-header-missing,
.card-header-not-eligible,
.card-header-useful {
  padding-left: 20px;
}

.card-header span {
  display: inline-block;
  vertical-align: bottom;
  padding-left: 20px;
  font-size: 1.4rem;
}

.card-header-eligible::before {
  content: '';
  position: absolute;
  left: 12px;
  top: 10px;
  width: 20px;
  height: 20px;
  display: block;
  background: url('/' + $service + '/tick.svg');
}

.card-header-missing::before {
  content: '';
  position: absolute;
  left: 12px;
  top: 10px;
  width: 20px;
  height: 20px;
  display: block;
  background: url('/' + $service + '/question.svg');
}

.card-header-useful::before {
  content: '';
  position: absolute;
  left: 12px;
  top: 10px;
  width: 20px;
  height: 20px;
  display: block;
  background: url('/' + $service + '/book.svg') no-repeat;
}

.card-header-not-eligible::before {
  content: '';
  position: absolute;
  left: 12px;
  top: 10px;
  width: 20px;
  height: 20px;
  display: block;
  background: url('/' + $service + '/cross.svg');
}

.card-tags {
  padding-top: 10px;
  margin: 0 -5px;
}

.card-tag {
  border-radius: 15px;
  background-color: $gray-light;
  font-size: 1.2rem;
  margin: 0 2.5px 5px 2.5px;
  padding: 5px 10px;
  text-align: center;
  display: inline-block;
}

.card-tag--fam {
  background-color: $brand-primary;
  color: white;
}

.card-eligibility-indicators {
  right: 0;
  position: absolute;
  top: -40px;
  padding: 5px 10px 5px 10px;
  background: $gray-med-light;
  font-size: 1.3rem;
  text-decoration: none;
  display: inline-block;
  text-align: right;

  @include media-breakpoint-down(xs) {
    top: -8px;
    left: -8px;
    display: block;
    width: calc(100% + 16px);
    position: relative;
  }
}

.card-eligibility-indicators span {
  text-decoration: none;
  display: inline-block;
  margin: 5px;
  min-height: 17px;
  text-align: center;
}

.card-eligibility-indicators span.card-eligibility-text {
  width: 100%;
  display: block;
  margin-top: 8px;
}


