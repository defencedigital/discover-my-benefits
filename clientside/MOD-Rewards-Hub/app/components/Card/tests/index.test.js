import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Card from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  subtitle: 'subtitle',
  text: 'text',
  image: 'image',
  link: 'link',
  openLinkInNewWindow: true,
  className: 'className',
  status: 1,
  onClick: () => {},
});

describe('Card component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<Card {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
