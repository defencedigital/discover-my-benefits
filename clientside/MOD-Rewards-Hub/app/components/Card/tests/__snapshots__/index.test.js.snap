// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `ShallowWrapper {}`;
